src/EFCore.SqlServer/Migrations/SqlServerMigrationsSqlGenerator.cs

```
@@ -117,6 +117,10 @@ protected override void Generate(MigrationOperation operation, IModel model, Mig
             MigrationCommandListBuilder builder,
             bool terminate)
         {
+            if (!terminate && operation.Comment != null)
+            {
+                throw new ArgumentException($"When generating migrations SQL for {nameof(AddColumnOperation)}, can't produce unterminated SQL with comments");
```

In SQLite, [we Debug.Fail() but let it slide](https://github.com/aspnet/EntityFrameworkCore/blob/d300fc2670ae19ee383bc604ab69e9ef647dae38/src/EFCore.Sqlite.Core/Migrations/SqliteMigrationsSqlGenerator.cs#L190) in Release.

---

But at least here we now for sure that the SQL will come out totally mangled, no? So better to throw an informative exception?

---

I mean... they *could* compose WITH RECOMPILE onto the last EXEC call 😜

---

Wow, OK :) Not sure what it would mean to do that to sp_addextendedproperty... In any case it would only work on the last statement, and only if comments were provided...

Basically I can't see the non-terminating scenario being much use when multiple statements are being generated - especially when the really significant one is the first...

---

I'm totally fine throwing 😄 (just pointing out an existing pattern used elsewhere)